lib/segment/src/vector_storage/raw_scorer.rs

```diff
@@ -915,16 +916,22 @@ where
     }
 
     fn peek_top_all(&self, top: usize) -> Vec<ScoredPointOffset> {
-        let scores = (0..self.point_deleted.len() as PointOffsetType)
+        let point_ids = (0..self.point_deleted.len() as PointOffsetType)
             .take_while(|_| !self.is_stopped.load(Ordering::Relaxed))
             .filter(|point_id| self.check_vector(*point_id))
-            .map(|point_id| {
-                let point_id = point_id as PointOffsetType;
-                ScoredPointOffset {
+            .chunks(64); // batch points to leverage storage sequential access
+
+        let scores = point_ids.into_iter().flat_map(|point_ids| {
+            let point_ids: Vec<PointOffsetType> = point_ids.collect();
```

Comment:
might be possible to avoid collect here?

Reply:
One constraint is checking that the ids are consecutive to pick the batching strategy. This requires a full traversal.